### ISSUE TITLE
How to combine unique and stategy merges

### ISSUE THREAD

**Issue by llaski** [survivejs/webpack-merge#119]
I can't seem to figure out how combine the unique and strategy merge options. I want to be able to prepend the results of a plugins array and then only keep unique plugin instances based on the constructor name. The default behavior of the unique function is to keep the first unique instance and I was the latter unique instance essentially. Example below:

```
configA = {
   plugins: [
        new MiniCssExtractPlugin({
            filename: 'main.[contenthash].css',
        }),
}

configB = {
   plugins: [
        new MiniCssExtractPlugin({
            filename: 'main.css',
        }),
}

merge({
      customizeArray: merge.unique(
        'plugins',
        ['MiniCssExtractPlugin'],
        plugin => plugin.constructor && plugin.constructor.name
      )
    })(configA, configB)
```

This will result in the `MiniCssExtractPlugin` plugin instance from configA being kept, and I want to keep the one from configB basically.

**Comment by chinesedfan** [comment]
The provided `merge.unique` always keeps the first one, but you can change it a bit as your customized unique function. Like,

```
 import { differenceWith } from 'lodash';
 function mergeUnique(key, uniques, getter = a => a) {
   return (a, b, k) => (
     k === key && [
-      ...a,
+      ...b,
       ...differenceWith(
-        b, a, item => uniques.indexOf(getter(item)) >= 0
+        a, b, item => uniques.indexOf(getter(item)) >= 0
       )
     ]
   );

merge({
      customizeArray: mergeUnique( // use your customized version
        'plugins',
        ['MiniCssExtractPlugin'],
        plugin => plugin.constructor && plugin.constructor.name
      )
    })(configA, configB)
```

**Comment by thibaut-singlefile** [comment]
I also expected merge unique to apply precedence to the item on the right light the rest of the merge package.

**Comment by bebraw** [comment]
There's a fix for this in #135 (v5). I changed the precedence there to pick the last first as that's the best option given the way webpack configurations are structured.
